src/Raven.Server/ServerWide/Maintenance/ClusterMaintenanceWorker.cs

```diff
@@ -145,13 +145,13 @@ private IEnumerable<(string name, DatabaseStatusReport report)> CollectDatabaseI
                             foreach (var index in indexStorage.GetIndexes())
                             {
                                 var stats = index.GetIndexStats(context);
-
-                                report.LastIndexStats.Add(index.Name, new DatabaseStatusReport.ObservedIndexStatus
+                                //We might have old version of this index with the same name
```

Comment:
I have changed it to run over indexes by name.
Just for completion, the way this happened was because we inserted a faulty index under the same name as the original when we created it.